server/server.go

```diff
@@ -84,6 +86,28 @@ func Server(ver string) *chi.Mux {
 	if err != nil {
 		logger.Error("failed to read/process configuration", zap.Error(err))
 	}
+
+	// check if logPath is provided, if yes, create a log file and write logs to it
+	if conf.LogPath != "" {
+		logPath := fmt.Sprintf("%s.log", conf.LogPath)
+		logFile, err := os.Create(logPath)
+		if err != nil {
+			logger.Error("failed to create log file", zap.Error(err))
+		}
+		defer logFile.Close()
+
+		fileSyncer := zapcore.AddSync(logFile)
+
+		core := zapcore.NewCore(
```

Comment:
Yes
If the `conf.LogPath` is empty, it will just log it on the console, but
If the `conf.LogPath` is not empty, it will write the logs in the file name given.

I am adding a demo script which follow the same procedure as written here for testing 
```
package main

import (
	"fmt"
	"os"
	"time"

	"go.uber.org/zap"
	"go.uber.org/zap/zapcore"
)

func main() {
	logConf := zap.NewDevelopmentConfig()
	logConf.Level = zap.NewAtomicLevelAt(zap.InfoLevel)
	logger, err := logConf.Build()
	if err != nil {
		panic(err)
	}
	defer logger.Sync()

	logPath := "" // if given a path, will write to file 
	// logPath = "test"

	if logPath != "" {
		logPath := fmt.Sprintf("%s.log", logPath)
		logFile, err := os.Create(logPath)
		if err != nil {
			logger.Error("failed to create log file", zap.Error(err))
		}
		defer logFile.Close()

		fileSyncer := zapcore.AddSync(logFile)

		core := zapcore.NewCore(
			zapcore.NewConsoleEncoder(logConf.EncoderConfig),
			fileSyncer,
			logConf.Level,
		)

		logger = zap.New(core)
	}

	logger.Info("hello world")
	time.Sleep(1 * time.Second)
	logger.Error("Goodbye world")
}
```

Hope this helps

Reply:
Got it. just tried this locally. It works perfectly. LGTM